docs/RFCS/sql_privileges.md

```diff
@@ -0,0 +1,208 @@
+	owner, or `GRANT OPTION` on revoked privileges. Complex rules for grantees.
+* cockroach: `GRANT` on affected object.
+
+`SELECT`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/select.html):
+  `SELECT` (doc seems incomplete).
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-select.html):
+  `SELECT` if tables are read. Also requires `UPDATE` when using `FOR UPDATE`.
+* cockroach: `SELECT` if tables are read.
+
+`INSERT`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/insert.html):
+  `INSERT`. `UPDATE` or `SELECT` for `ON DUPLICATE KEY UPDATE`.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-insert.html):
+  `INSERT`. `SELECT` if using `RETURNING`.
+* cockroach: `INSERT` on table. `SELECT` if using `SELECT` clause on tables.
+
+`DELETE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/delete.html):
+	`DELETE` on table. `SELECT` if `WHERE` clause operates on tables.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-delete.html):
+	`DELETE` on table. `SELECT` if `USING` or `WHERE` clauses operate on tables.
+* cockroach: `DELETE` on table. `SELECT` if `WHERE` clause operates on tables.
+
+`UPDATE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/update.html):
+	`UPDATE`. Also `SELECT` for `WHERE` clause with table.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-update.html):
+	`UPDATE`. Also `SELECT` for `WHERE` clause with table.
+* cockroach: `UPDATE`. Also `SELECT` for `WHERE` clause with table.
+
+`TRUNCATE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/truncate-table.html):
+	`DROP` on table.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-truncate.html):
+  `TRUNCATE` privilege.
+* cockroach: `DROP` on table.
+
+`SHOW`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/show.html):
+  shows objects the user has privileges on.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-show.html):
+  only `SHOW name` is supported. No privileges required.
+* cockroach: no privileges required (we may wish something closer to mysql).
+
+`SET`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/set.html):
+  some require superuser, some cannot be changed after session start.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-set.html):
+  unknown. Probably no privileges.
+* cockroach: session variables can be changed at will. Privileges for
+  other types of variables will be revisited.
+
+## Storing privileges
+
+Privileges are stored in the `DatabaseDescriptor` and `TableDescriptor` using
+a list sorted by user.
+This gives the planner access to privileges after a descriptor lookup.
+
+Future improvements include gossiping descriptors [#1743](https://github.com/cockroachdb/cockroach/pull/1743)
+and potentially using a real table.
+
+# Drawbacks
+
+Postgres and mysql both have global user tables. We do not require users
+to be in the user config, only to be properly authenticated. This means
+that we cannot have user-defined super users.
```

Comment:
I think we will ultimately want a "users" table or config. We sort of have one, but it's only for the website login, it's not required. The major reason not to have it right now it performance (it's yet another lookup), but we'll need a story for privileges, accounting, and zones other than storing them all in the descriptor. At that point, the users config can follow and become required.